### PR TITLE
Add interaction support for View.message and add disable_on_timeout to Views

### PR DESCRIPTION
**Pull request by ConchDev**
## Summary

This pull requests defines a message for Views when sent as an interaction response. It also adds a new `disable_on_timeout` argument to ui.View to automatically disable all View components on the view's timeout.

## Information

- [x] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.

### CODE REVIEW

**Comment by Lulalaby** [comment]
Thanks conchy ♥️

**Review by baronkobama — APPROVED** [review]
LGTM

File: discord/ui/view.py [hunk 1]
```diff
@@ -341,7 +345,10 @@ async def on_timeout(self) -> None:
 
         A callback that is called when a view's timeout elapses without being explicitly stopped.
         """
-        pass
+        if self.disable_on_timeout:
+            if self._message:
+                self.disable_all_items()
+                await self._message.edit(view=self)
```
Comment (Dorukyum) [inline discord/ui/view.py:351]: I don't really think it's a good idea to have the lib edit the message by default

Comment (ConchDev) [reply discord/ui/view.py:351]: It doesn't. The disable_on_timeout kwarg defaults to False.

**Review by Middledot — APPROVED** [review]
lgtm

**Pull request merged**